src/Type/TypeCombinator.php

```diff
@@ -638,11 +638,13 @@ public static function intersect(Type ...$types): Type
 			$topLevelUnionSubTypes = [];
 			$innerTypes = $type->getTypes();
 			usort($innerTypes, $sortTypes);
+			$slice1 = array_slice($types, 0, $i);
```

Comment:
Hi,

Just out of curiosity, why creating intermediary variables here?

Reply:
Oh you're right, for some reason I haven't seen the `foreach` loop ! Well done.

Reply:
@neclimdul I feel this `array_slice` change is a no-brainer and should be submitted as a seperate PR.
the other change breaks the tests atm, so needs further inspection

Reply:
I agree, please send a microoptimization PR, then we'll discuss the rest.